File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [1]*len(nums)
        right = [1]*len(nums)
        n = len(nums)
        ans = []
        left_i = 1
        right_i = 1
        for i in range(n):
            left[i] = left_i
            left_i *= nums[i] 
            right[n-i-1] = right_i
            right_i *= nums[n-i-1]
        
        for i in range(n):
            ans.append(left[i]*right[i])

        return ans

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if(!root) return ans;
        queue<TreeNode*> q;
        q.push(root);
        q.push(NULL);
        vector<int> curr;
        while(!q.empty()){
            TreeNode* front = q.front();
            q.pop();
            if(!front){
                ans.push_back(curr);
                curr = {};
                if(!q.empty()) q.push(NULL);
            }
            else{
                curr.push_back(front->val);
                if(front->left) q.push(front->left);
                if(front->right) q.push(front->right);
            }
        }
        return ans;
    }
};
